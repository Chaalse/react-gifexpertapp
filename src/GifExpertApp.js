import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    // const categorias =['Star Wars', 'Samurai Jack', 'Vengadores'];
    const [categorias, setCategorias] = useState(['Star Wars']);

    // const handleAdd = () => {
    //     setCategorias( categorias => [...categorias, 'LOTR']);
    // }
    return (
        <>
        <h2>GifExpertApp</h2>
        <h3 >#BeFrikiMyFriend</h3>
        <AddCategory setCategorias = { setCategorias}/>
        <hr/>
        <ol>
            {
                categorias.map(categoria => (
                    <GifGrid 
                    key = {categoria}
                    categoria = { categoria}
                    />
                ))
            }
        </ol>

        </>

    )
}


export default GifExpertApp;