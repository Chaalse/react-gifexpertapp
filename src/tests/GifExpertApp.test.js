import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas de la funcionalidad de GifExpertApp', () => {
    
    
    test('Debe mostrar la aplicaciÃ³n', () => {
    
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar las categorias', () => {
        const categorias = ['Samurai Jack', 'Gargoyles'];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categorias }/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);

    });
    
    
})
