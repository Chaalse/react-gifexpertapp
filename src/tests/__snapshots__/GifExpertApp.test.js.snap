// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de la funcionalidad de GifExpertApp Debe mostrar la aplicaci√≥n 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <h3>
    #BeFrikiMyFriend
  </h3>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    ;
  </ol>
</Fragment>
`;

exports[`Pruebas de la funcionalidad de GifExpertApp Debe mostrar las categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <h3>
    #BeFrikiMyFriend
  </h3>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Samurai Jack"
      key="Samurai Jack"
    />
    <GifGrid
      categoria="Gargoyles"
      key="Gargoyles"
    />
    ;
  </ol>
</Fragment>
`;
